cpp/src/strings/slice.cu

```diff
@@ -92,14 +124,50 @@ struct substring_fn {
     char* d_buffer  = d_chars ? d_chars + d_offsets[idx] : nullptr;
     auto itr        = begin;
     while (step > 0 ? itr < end : end < itr) {
-      bytes += bytes_in_char_utf8(*itr);
-      if (d_buffer) d_buffer += from_char_utf8(*itr, d_buffer);
+      if (d_buffer) {
+        d_buffer += from_char_utf8(*itr, d_buffer);
+      } else {
+        bytes += bytes_in_char_utf8(*itr);
+      }
```

Comment:
Was this a bug before? Since `bytes` is computed in both situations?

Reply:
No. The `bytes` value is not actually used in the second path so I am just saving a few instructions.